tests/integration/test_ec2.py

```diff
@@ -222,6 +222,52 @@ def test_describe_vpn_gateways_filter_by_vpc(self):
         ec2.delete_vpn_gateway(VpnGatewayId=gateway_id)
         ec2.delete_vpc(VpcId=vpc_id)
 
+    def test_describe_vpc_endpoints_with_filter(self):
+
+        ec2 = self.ec2_client
+        vpc = ec2.create_vpc(CidrBlock="10.0.0.0/16")
+        vpc_id = vpc["Vpc"]["VpcId"]
+
+        # test filter of Gateway endpoint services
+        vpc_endpoint_gateway_services = ec2.describe_vpc_endpoint_services(
+            Filters=[
+                {"Name": "service-type", "Values": ["Gateway"]},
+            ],
+        )
+
+        self.assertEqual(200, vpc_endpoint_gateway_services["ResponseMetadata"]["HTTPStatusCode"])
+        services = vpc_endpoint_gateway_services["ServiceNames"]
+        self.assertEqual(2, len(services))
+        self.assertTrue("com.amazonaws.us-east-1.dynamodb" in services)
+        self.assertTrue("com.amazonaws.us-east-1.s3" in services)
+        # test filter of Interface endpoint services
+        vpc_endpoint_interface_services = ec2.describe_vpc_endpoint_services(
+            Filters=[
+                {"Name": "service-type", "Values": ["Interface"]},
+            ],
+        )
+
+        self.assertEqual(200, vpc_endpoint_interface_services["ResponseMetadata"]["HTTPStatusCode"])
+        services = vpc_endpoint_interface_services["ServiceNames"]
+        self.assertTrue(len(services) > 0)
+        self.assertTrue("com.amazonaws.us-east-1.dynamodb" in services)
```

Comment:
Nit: We could use `assertIn(..)` here - not critical, though, as we're soon going to migrate these tests from `unittest` to `pytest` anyway. 👍